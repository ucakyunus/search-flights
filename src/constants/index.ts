export enum Currency {
  TRY = 'TRY',
}

export enum FareCategories {
  BUSINESS = 'BUSINESS',
  ECONOMY = 'ECONOMY',
}

export enum BrandCode {
  ECO_FLY = 'ecoFly',
  EXTRA_FLY = 'extraFly',
  PRIME_FLY = 'primeFly',
}

export enum Status {
  AVAILABLE = 'AVAILABLE',
  ERROR = 'ERROR',
}

export enum ListSortTypes {
  PriceAscending = 'PriceAscending',
  PriceDescending = 'PriceDescending',
  DepartureTimeAscending = 'DepartureTimeAscending',
  DepartureTimeDescending = 'DepartureTimeDescending',
}

export enum ListSortBy {
  DepartureTime = 'DepartureTime',
  Price = 'Price',
}

export enum ResultTypes {
  Success = 'Success',
  Failure = 'Failure',
}

export enum CounterActions {
  Increment = 'INCREMENT',
  Decrement = 'DECREMENT',
}
