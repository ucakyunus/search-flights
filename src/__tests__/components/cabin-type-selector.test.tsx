import { render } from '@testing-library/react';
import {
  CabinTypeSelector,
  CabinTypeSelectorProps,
} from '@/components/cabin-and-passenger/cabin-type-selector';
import { FareCategories } from '@/constants';

const mockProps: CabinTypeSelectorProps = {
  value: FareCategories.ECONOMY,
  onChange: jest.fn(),
};

test('CabinTypeSelector matches snapshot', () => {
  const { asFragment } = render(<CabinTypeSelector {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
