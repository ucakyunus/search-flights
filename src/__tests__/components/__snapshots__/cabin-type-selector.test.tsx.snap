// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabinTypeSelector matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="cabin-type"
    class="MuiFormGroup-root MuiFormGroup-row MuiRadioGroup-root MuiRadioGroup-row css-cis11t-MuiFormGroup-root"
    role="radiogroup"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      color="primary"
    >
      <span
        class="MuiButtonBase-root MuiRadio-root MuiRadio-colorInfo PrivateSwitchBase-root MuiRadio-root MuiRadio-colorInfo Mui-checked MuiRadio-root MuiRadio-colorInfo css-3ohrad-MuiButtonBase-root-MuiRadio-root"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-j8yymo"
          name="cabin-type"
          type="radio"
          value="ECONOMY"
        />
        <span
          class="css-1qiat4j"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z8nmqa-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Economy Class
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiButtonBase-root MuiRadio-root MuiRadio-colorInfo PrivateSwitchBase-root MuiRadio-root MuiRadio-colorInfo MuiRadio-root MuiRadio-colorInfo css-3ohrad-MuiButtonBase-root-MuiRadio-root"
      >
        <input
          class="PrivateSwitchBase-input css-j8yymo"
          name="cabin-type"
          type="radio"
          value="BUSINESS"
        />
        <span
          class="css-1qiat4j"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Business Class
      </span>
    </label>
  </div>
</DocumentFragment>
`;
